applications/IgaApplication/custom_elements/shell_kl_discrete_element.cpp

```diff
@@ -0,0 +1,512 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                     Kratos default license: kratos/IGAStructuralMechanicsApplication/license.txt
+//
+//  Main authors:    Tobias Tescheamacher
+//                   Riccardo Rossi
+//
+
+
+// System includes
+#include "includes/define.h"
+#include "utilities/math_utils.h"
+#include "geometries/geometry.h"
+
+// External includes
+
+// Project includes
+#include "custom_elements/shell_kl_discrete_element.h"
+
+// Application includes
+#include "iga_application.h"
+#include "iga_application_variables.h"
+
+namespace Kratos
+{
+    //************************************************************************************
+    //************************************************************************************
+    void ShellKLDiscreteElement::CalculateAll(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo,
+        const bool CalculateStiffnessMatrixFlag,
+        const bool CalculateResidualVectorFlag
+    )
+    {
+        KRATOS_TRY
+        // definition of problem size
+        const unsigned int number_of_nodes = GetGeometry().size();
+        unsigned int mat_size = number_of_nodes * 3;
+
+        //set up properties for Constitutive Law
+        ConstitutiveLaw::Parameters Values(GetGeometry(), GetProperties(), rCurrentProcessInfo);
+
+        Values.GetOptions().Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
+        Values.GetOptions().Set(ConstitutiveLaw::COMPUTE_STRESS);
+        Values.GetOptions().Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR);
+
+        //resizing as needed the LHS
+        if (CalculateStiffnessMatrixFlag == true) //calculation of the matrix is required
+        {
+            if (rLeftHandSideMatrix.size1() != mat_size)
+                rLeftHandSideMatrix.resize(mat_size, mat_size);
+            noalias(rLeftHandSideMatrix) = ZeroMatrix(mat_size, mat_size); //resetting LHS
+        }
+        //resizing as needed the RHS
+        if (CalculateResidualVectorFlag == true) //calculation of the matrix is required
+        {
+            if (rRightHandSideVector.size() != mat_size)
+                rRightHandSideVector.resize(mat_size);
+            rRightHandSideVector = ZeroVector(mat_size); //resetting RHS
+        }
+
+        //reading in of integration weight, shape function values and shape function derivatives
+        double integration_weight = this->GetValue(INTEGRATION_WEIGHT);
```

Comment:
This value of `integration_weight` is never used. It can be declared below where it is reassigned a different value.